### ISSUE TITLE
Add Test plan report status dialog (P1)

### ISSUE THREAD

**Issue by lolaodelola** [w3c/aria-at-app#649]
_Taken from #648._

This page or dialog shows detailed status of AT/browser reports for a given version of a given test plan. The AT/browser combinations are limited to the AT covered by the plan and the most recent version of each in-scope browser included in the system browser list.

It includes:

- Links to completed reports
- Status of in-progress reports
- Information about missing reports
- Whether a specific report is required for the current phase
- The data is presented in a table captioned "PHASE_NAME TEST_PLAN_NAME test Plan Reports". Example: "Draft Alert Test Plan Reports".

The table has columns for:

1. required?: Shows "Yes" or "No"
2. AT: Shows AT Name and version
3. Browser: Shows browser name and version
4. Report Status: Shows:
    - If the report is complete, a link to the report with text "Report Completed on DATE"
    - If report is missing, the string "Missing", and if user is admin, button "Add to Test Queue"
    - If report is not complete and not missing and not in test queue, the string "None", and if user is admin, "Add to Test Queue" button
    - If report is in the test queue, the string "X% complete by Y testers with Z conflicts"
    - Note: Percent complete = NUMBER_COMPLETED_TESTS_BY_ALL_TESTERS / (NUMBER_ASSIGNED_TESTERS * NUMBER_TESTS_IN_PLAN) * 100

The table is sorted by required Yes, then required No, then AT, then browser.

**Comment by lolaodelola** [comment]
This needs to be split into individual tasks and prioritised, for now it's a P2 until we can discuss it.

**Comment by lolaodelola** [comment]
Hi @mcking65, Do you have a preference between page & dialogue? It'd be good for you and @isaacdurazo to coordinate on what the UI needs for this dialogue/page is, also it'd be helpful to know what the prioritisation of this work is, I've put it at P0 for now.

**Comment by ccanash** [comment]
Updating description per latest update in #648  @lolaodelola  @mcking65  from latest update it seems now we know it will be a dialogue. 


<h4 dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 14px; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Test plan report status dialog Description</h4><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">This dialog shows detailed status of AT/browser reports for a single version of a single test plan.</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The label on the button that opens the dialog provides information about the current report status for the test plan. The button has one of the following labels.</p>

Button Label | Condition
-- | --
Required Reports Not Started | For every AT/browser pair that requires a report, the report is neither complete nor in the test queue.
Some Required Reports Missing | At least one of the AT/browser pairs that requires a report neither has a complete report nor has a run in the test queue. At the same time, at least one of the AT/browser pairs that requires a report either has a complete report or has a run in the test queue.
Required Reports In Progress | At least one AT/browser pair that requires a report does not have a complete report and is in the test queue. All other AT/browser pairs that require a report are either complete or are in the test queue.
Required Reports Complete | All AT/browser pairs that require a report have a complete report.

<p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The dialog title is: "Report Status for TEST_PLAN_NAME Test Plan"</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The dialog content includes a note and a table.</p><ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box; margin-left: 0px;">Note content:<ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 0px;"><li style="box-sizing: border-box; margin-left: 0px;">"Note: This plan is in the PHASE_NAME phase. N AT/browser pairs require reports in this phase."</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">Example: "This plan is in the Draft Review phase. 3 AT/browser pairs require reports in this phase."</li></ul></li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">The table is captioned:<ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 0px;"><li style="box-sizing: border-box; margin-left: 0px;">"Reports for PHASE_NAME TEST_PLAN_NAME Test Plan"</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">Example: "Reports for Draft Alert Test Plan".</li></ul></li></ul><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The table columns are:</p><ol dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box; margin-left: 0px;">required?: Shows "Yes" or "No"</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">AT:<ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 0px;"><li style="box-sizing: border-box; margin-left: 0px;">If phase is draft, shows AT name.</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">If phase is recommended, Shows AT Name and version.</li></ul></li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">Browser:<ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 0px;"><li style="box-sizing: border-box; margin-left: 0px;">If phase is draft, shows browser name.</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">If phase is recommended, shows browser name and version.</li></ul></li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">Report Status:<ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 0px;"><li style="box-sizing: border-box; margin-left: 0px;">If the report is complete, shows a link to the report with text "Report Completed on DATE"</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">If report is missing, shows the string "Missing", and if user is admin, button "Add to Test Queue"</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">If report is not complete and not missing and not in test queue, shows the string "None", and if user is admin, "Add to Test Queue" button</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">If report is in the test queue, shows the string "X% complete by Y testers with Z conflicts"</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">Note: Percent complete = NUMBER_COMPLETED_TESTS_BY_ALL_TESTERS / (NUMBER_ASSIGNED_TESTERS * NUMBER_TESTS_IN_PLAN) * 100</li></ul></li></ol><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The table is sorted by required Yes, then required No, then AT, then browser.</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The rows that appear in the table are determined as follows.</p><ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box; margin-left: 0px;">If the phase is draft, there is one row for each valid AT/browser pair for all AT covered by the plan.</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;">TODO: If the phase is recommended, the algorithm for determining what rows to put in the table will be described in the next update of this specification.</li></ul>

**Comment by isaacdurazo** [comment]
Here's the mockup for this dialog.

Keep in mind that this is not a real scenario since we wouldn't have 4 required reports for a Draft Test plan. I added an extra row just so I could show all the different Report statuses.

@mcking65 This table is pretty simple in terms of Visual Design. The Required, AT, and Browser columns display all plain text. The Report Status uses our existing styling for secondary buttons when we have the "Add to Test Queue" button and the statuses strings reuse the little circle icon I propose in issue #648.

![ARIA-AT - Data Management – Test plan report status dialog](https://github.com/w3c/aria-at-app/assets/1379244/043179ce-af7e-4588-bca9-c1e5a93e2345)

**Comment by ccanash** [comment]
Hi @mcking65 , we plan on working on this on Wednesday. Would it be possible for you to review the design beforehand?

**Comment by mcking65** [comment]
Hi @ccanash, I don't see anything for me to review.

**Comment by isaacdurazo** [comment]
@mcking65 [This is the comment](https://github.com/w3c/aria-at-app/issues/649#issuecomment-1633171161) with the proposed dialog.

**Comment by mcking65** [comment]
@isaacdurazo wrote:

> Here's the mockup for this dialog.
> 
> Keep in mind that this is not a real scenario since we wouldn't have 4 required reports for a Draft Test plan. I added an extra row just so I could show all the different Report statuses.
> 
> @mcking65 This table is pretty simple in terms of Visual Design. The Required, AT, and Browser columns display all plain text. The Report Status uses our existing styling for secondary buttons when we have the "Add to Test Queue" button and the statuses strings reuse the little circle icon I propose in issue #648.

That sounds good to me.

> 
> ![ARIA-AT - Data Management – Test plan report status dialog](https://user-images.githubusercontent.com/1379244/253091718-043179ce-af7e-4588-bca9-c1e5a93e2345.png)

I am assuming the content in the image matches what is described in the top comment. Is a version of this viewable in the sandbox now?

**Comment by mcking65** [comment]
Tested in staging. Overall, this dialog looks great! Thank you.

There are two unmet requirements and two bugs. I raised separate issues for them so we can close this issue after deployment.

Requirements:

>If phase is recommended, Shows AT Name and version. 
>If phase is recommended, shows browser name and version. 

The rationale for showing versions is related to #792. The version numbers are not especially helpful until we can actually generate reports for specific versions. So, we can let #792 and the issues it depends on track development of that capability.

The two bugs are documented in #796. Those are not blocking issues, so we can consider this issue complete as soon as we deploy.
